### PR TITLE
Use thinc-apple-ops>=0.1.0.dev0 with `apple` extras

### PR DESCRIPTION
**Pull request by danieldk**
## Description

This change ensures that installing spaCy with `apple` uses thinc-apple-ops version 0.1.0.dev0 or later, which has the proper CBLAS passthrough. Also test with at least this version.

### Types of change

Enhancement

## Checklist

- [x] I confirm that I have the right to submit this contribution under 
the project's MIT license.
- [ ] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they 
do, I've added all required information.

### CODE REVIEW

File: .github/azure-steps.yml [hunk 1]
```diff
@@ -111,7 +111,7 @@ steps:
     condition: eq(variables['python_version'], '3.8')
 
   - script: |
-      ${{ parameters.prefix }} python -m pip install thinc-apple-ops
+      ${{ parameters.prefix }} python -m pip install "thinc-apple-ops>=0.1.0.dev0"
       ${{ parameters.prefix }} python -m pytest --pyargs spacy
     displayName: "Run CPU tests with thinc-apple-ops"
     condition: and(startsWith(variables['imageName'], 'macos'), eq(variables['python.version'], '3.9'))
```
Comment (adrianeboyd) [inline .github/azure-steps.yml:117]: I think this is out-of-date:

```suggestion
    condition: and(startsWith(variables['imageName'], 'macos'), eq(variables['python.version'], '3.10'))
```

File: .github/azure-steps.yml [hunk 2]
```diff
@@ -111,7 +111,7 @@ steps:
     condition: eq(variables['python_version'], '3.8')
 
   - script: |
-      ${{ parameters.prefix }} python -m pip install thinc-apple-ops
+      ${{ parameters.prefix }} python -m pip install "thinc-apple-ops>=0.1.0.dev0"
```
Comment (adrianeboyd) [inline .github/azure-steps.yml:114]: The only slightly more general option might be to use `--pre` here instead?

Comment (danieldk) [reply .github/azure-steps.yml:114]: I didn't know about that option. So, then this should be

```
      ${{ parameters.prefix }} python -m pip install --pre "thinc-apple-ops>=0.1.0"
```

?

Comment (adrianeboyd) [reply .github/azure-steps.yml:114]: Just

```
python -m pip install --pre thinc-apple-ops
```

Comment (adrianeboyd) [reply .github/azure-steps.yml:114]: I'm not sure if we'd have cases where we specifically wouldn't want `--pre`, but it's probably okay as a default here for this test?

Comment (danieldk) [reply .github/azure-steps.yml]: Yeah, I think for CI this should be completely safe. I've added a commit to use `--pre`.

**Pull request merged**